### PR TITLE
Bug 1634745 - Export ping request for use by Rust consumers

### PR DESCRIPTION
**Pull request by badboy**
Also includes minor other fixes.

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt [hunk 1]
```diff
@@ -43,14 +43,18 @@ internal class UploadBody(
     }
 }
 
-@Structure.FieldOrder("tag", "upload")
```
Comment (Dexterp37) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:46]: Why was this removed?

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt [hunk 2]
```diff
@@ -43,14 +43,18 @@ internal class UploadBody(
...
 internal open class FfiPingUploadTask(
     // NOTE: We need to provide defaults here, so that JNA can create this object.
     @JvmField var tag: Byte = UploadTaskTag.Done.ordinal.toByte(),
     @JvmField var upload: UploadBody = UploadBody()
 ) : Union() {
     class ByValue : FfiPingUploadTask(), Structure.ByValue
 
+    init {
+        // Initialize to be the `tag`-only variant
+        setType("tag")
```
Comment (Dexterp37) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:55]: Interesting. Why was this working before?

Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:55]: Because that field is valid in both variants (it's also the first 8 bits in the `UploadBody`, therefore the first 8 bits of the union are always an integer).
This just makes it more explicit, which IMO helps us still understand this code in 2, 4 and 8 weeks.

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt (hunk 1 above)
Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:46]: See the commit: a union hardly as an order, only one field is "active" at a time.
(The underlying `Union` inherits from `Structure` and just has a default implementation for `FieldOrder` to satisfy the class hierarchy)

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt (hunk 2 above)
Comment (Dexterp37) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt:55]: Interesting! Thanks for clarifying!
